payments/models.py

```diff
@@ -85,6 +182,17 @@ def establish_basket(request, integrated_system: IntegratedSystem):
         ),
     ]
 
+    def apply_discount(self, discount_id):
+        """
```

Comment:
nit: can we use the built in relationship stuff here? This is already a `ManyToMany` so we have `discounts.add()`. Otherwise this should be renamed to indicate that it's taking in an ID specifically and the doc updated.